frame/contracts/src/exec.rs

```diff
@@ -537,13 +564,16 @@ where
 
 	fn instantiate(
 		&mut self,
-		code_hash: &CodeHash<T>,
+		code_hash: CodeHash<T>,
 		endowment: BalanceOf<T>,
 		gas_meter: &mut GasMeter<T>,
 		input_data: Vec<u8>,
 		salt: &[u8],
 	) -> Result<(AccountIdOf<T>, ExecReturnValue), ExecError> {
-		self.ctx.instantiate(endowment, gas_meter, code_hash, input_data, salt)
+		let executable = E::from_storage(code_hash, &self.ctx.config.schedule)?;
+		let result = self.ctx.instantiate(endowment, gas_meter, &executable, input_data, salt)?;
+		executable.store();
```

Comment:
As mentioned earlier it  doesn't seem right to call `store()` which will check for the existence of the code first, which in this case we already know exists.

Reply:
See above comment. Maybe the name should be changed to be less eye brow rasing.